### PR TITLE
DOC: Cross-compilation

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/conda-forge/conda-forge.github.io/issues/1858

Provide some examples on how to adapt recipes for cross-compilation.

PR Checklist:

- [ ] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [ ] put any other relevant information below

### CODE REVIEW

File: src/maintainer/knowledge_base.rst [hunk 1]
```diff
@@ -279,6 +279,61 @@ A package that needs all five compilers would define
...
+        - python
+        - pip
+        - cython
+        - numpy
+      run:
+        - python
+        - {{ pin_compatible("numpy") }}
+
+There are more variations of this approach in the wild. So this is not meant to be exhaustive,
+but merely to provide a starting point with some guidelines. Please look at other recipes for more examples.
```
Comment (beckermr) [inline src/maintainer/knowledge_base.rst:335]: Is there a canned github search URL we can provide here?

Comment (jakirkham) [reply src/maintainer/knowledge_base.rst:335]: Suppose one could search for `build_platform` and/or `target_platform`. Maybe not perfect. Also may still need tweaking, but that could be a starting point.

Mostly was trying to get at the fact that some things use `pybind11`, `boost`, `swig`, or something else. R can be a bit different too. The general themes are the same, but different tweaks in other cases.

Comment (beckermr) [reply src/maintainer/knowledge_base.rst:335]: What about this one?

https://github.com/search?q=org%3Aconda-forge%20%22%5Bbuild_platform%20!%3D%20target_platform%5D%22&type=code

Comment (jakirkham) [reply src/maintainer/knowledge_base.rst:335]:
```suggestion
but merely to provide a starting point with some guidelines. Please look at `other recipes for more examples`_.

.. _other recipes for more examples: https://github.com/search?q=org%3Aconda-forge+path%3Arecipe%2Fmeta.yaml+%22%5Bbuild_platform+%21%3D+target_platform%5D%22&type=code
```

**Comment by beckermr** [comment]
Thank you!

**Comment by jakirkham** [comment]
Thanks Matt! 🙏

**Pull request merged**